### DISCUSSION TITLE
Publisher onboarding to VS Marketplace - feedback

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#238] [Extension Development QnA] [answered]
We are looking for any feedback regarding when you on-boarded to be a VS Code extension publishers on the VS Marketplace.
The process is mostly described here https://code.visualstudio.com/api/working-with-extensions/publishing-extension#publishing-extensions

While any feedback is very much appreciated here are some additional questions:
* How was the experience of creating an Azure Dev Ops account? 
* Did you already have an Azure Dev Ops account? 
* Would you prefer to be able to use some other account (for example GitHub)? 
* Do you use your Azure Dev Ops account for other functionality?
* How was the process of authenticating through `vsce` using Azure DevOps PAT tokens?

**Comment by RandomFractals** [comment]
I think publishing only with GitHub, without Azure dev ops account setup, would open it up to more devs trying to get into vscode extensions development without much cloud setup friction.

**Comment by eamodio** [comment]
I always found the Azure DevOps part confusing and a pain point in the setup, especially when having to deal with expired tokens. It's a disconnected experience from the marketplace and publishing as a whole.

Using a GitHub account imo would be better, mainly because I think more folks would already have it (I had to create the azdo stuff just for the extension, which was not a good experience).

But at the same time, I feel the particular account is a bit less of a problem if the whole flow was integrated and the mechanics were largely hidden from the user by the marketplace itself.

**Comment by alefragnani** [comment, marked as answer]
First of all, congrats for the initiative.

It's been a long time I created my publisher account to publish my first VS Code extensions, and I guess it wasn't even on Azure yet, but I have memories that it wasn't a complex process, but some _concepts_ were new to me and I felt a bit worried to not mess up 😆 . On the other hand, the instructions were clear and the only mess I did (and still do sometimes) was erroneous selection of the `Scope` for the PAT. And I agree with @eamodio about expired tokens. Maybe a _renew_ process would be easier. 

About the current documentation, I would suggest one thing: split the `Publishing Extensions` page in two, moving the whole `Create a publisher / PAT` sections to a separate page, so newcomers will easily identify that it needs to create a publisher first. Other than that, it is really good.

About the usage of Azure DevOps, I only use it for VS Code extensions publishing, nothing else.

About using _other account_ (like GitHub), I would agree, but _only_ if this means _remove the Azure DevOps piece from the process_, instead of replacing for another one. And, as you may know, some extension authors uses Gitlab or other services to host their extensions. 

TBH, I wonder why it is necessary to have an Azure DevOps account to publish extensions. Why can't it be a simpler solution like publishing a npm package or Atom plugins? Simply have an account in that marketplace/store and `publish` it there?

Thank you

**Comment by PEZ** [comment]
I've had tons of of problems with the Azure stuff. Sometimes it has been buggy, but mostly the Ux is from somewhere super alien to me. So I avoid Azure DevOps as much as I can. But every time my token expires it takes me all too much time to google and figure out how to update it.

That said, I remember the rest of the onboarding being blissful. Super easy to go from nothing to having an extension published.
